ocaml/xapi/message_forwarding.ml

```diff
@@ -2160,7 +2210,20 @@ functor
                 else
                   forward_to_access_srs ~local_fn ~__context ~vm forward_fn
             )
-        )
+        ) ;
+        let message_body =
+          Printf.sprintf "VM '%s' reverted to snapshot: %s"
+            (Db.VM.get_name_label ~__context ~self:vm)
+            (Db.VM.get_name_label ~__context ~self:snapshot)
+        in
+        let name, priority = Api_messages.vm_snapshot_reverted in
+        try
+          ignore
+            (Xapi_message.create ~__context ~name ~priority ~cls:`VM
+               ~obj_uuid:(Db.VM.get_uuid ~__context ~self:vm)
+               ~body:message_body
+            )
+        with _ -> ()
```

Comment:
There is quite a bit of repetition here, it feels like creating a function that is used in all these sites is worth it, something like:
```ocaml
let create_vm_message create_body message vm =
  let message_body = create_body () in
  let name, priority = Api_messages.vm_snapshot_reverted in
  try
    ignore
      (Xapi_message.create ~__context ~name ~priority ~cls:`VM
         ~obj_uuid:(Db.VM.get_uuid ~__context ~self:vm)
         ~body:message_body
      )
    with _ -> ()

...

let create_body () = 
  Printf.sprintf "VM '%s' reverted to snapshot: %s"
    (Db.VM.get_name_label ~__context ~self:vm)
    (Db.VM.get_name_label ~__context ~self:snapshot)
in
create_vm_message create_body Api_messages.vm_snapshot_reverted vm
```